ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js

```diff
@@ -77,6 +81,19 @@
             this.widget.addEventListener('focus', (e) => {
                 this.setActive();
             });
+            this.setupCharacterCounter();
+        }
+
+        setupCharacterCounter() {
+        	const counterSpan = this.getCounterSpan();
+        	const updateCharacterCounter = () => {
+        		const charCount = this.widget.value.length;
+        		if (counterSpan) {
+        			counterSpan.textContent = Granite.I18n.getMessage(charCount);
```

Comment:
for localizing the number at runtime, you can't use Granite.I18n. You will need to do what is done at https://github.com/adobe/aem-core-forms-components/blob/dev/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js#L1013